clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java

```diff
@@ -802,15 +815,19 @@ public void run() {
         controllerThreadId = Thread.currentThread().getId();
         try {
             processingCycle = true;
-            while(running)
+            while( isRunning() ) {
                 tick();
+            }
         } catch (InterruptedException e) {
             log.log(LogLevel.DEBUG, "Event thread stopped by interrupt exception: " + e);
         } catch (Throwable t) {
             t.printStackTrace();
             log.log(LogLevel.ERROR, "Fatal error killed fleet controller", t);
-            synchronized (monitor) { running = false; }
+            synchronized (monitor) { running.set(false); }
```

Comment:
Is this required when `running.set(false)` is always invoked in the `finally` block below?

Reply:
Since there is a System.exit on the line following it, it will never reach the finally block.
But if it does any good is another question. I just thought I would leave it there. I am not sure what happens after System.exit has been called. Since it is not a daemon thread it might block exit(). So I chose just leave it as it is.

Reply:
Good point, best to stay on the safe side.